src/main/java/baza/trainee/service/impl/ArticleServiceImpl.java

```
@@ -1,19 +1,26 @@
 package baza.trainee.service.impl;
 
 import baza.trainee.domain.model.Article;
+import baza.trainee.repository.ArticleRepository;
 import baza.trainee.service.ArticleService;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import java.util.Optional;
+
 @Service
 public class ArticleServiceImpl implements ArticleService {
-    /**
-     * Finds an existing article by given title.
-     *
-     * @param title title to get an existing article.
-     * @return {@link Article} object containing an existing article with its full content.
-     */
+
+    private final ArticleRepository articleRepository;
+
+    @Autowired
+    public ArticleServiceImpl(ArticleRepository articleRepository) {
+        this.articleRepository = articleRepository;
+    }
+
     @Override
-    public Article findByTitle(final String title) {
-        return null;
+    public Article findByTitle(String title) {
+        // Implement the logic to fetch the article by title from the repository
+        return articleRepository.findByTitle(title);
```

When the repository returns an Optional Article, you need to check whether it is empty. If it is, you should throw an exception.

---

fixed